### ISSUE TITLE
RCS0053 false positive when using top-level statements

### ISSUE THREAD

**Issue by xt0rted** [JosefPihrt/Roslynator#890]
**Product and Version Used**: 4.1.0

**Steps to Reproduce**: The full code triggering this can be [seen here](https://github.com/xt0rted/dotnet-run-script/blob/8400bdd604a6ef3f677946601f742d5a17a971df/src/Program.cs#L63-L76) and a branch with the error is here https://github.com/xt0rted/dotnet-run-script/pull/11.

```csharp
var parser = new CommandLineBuilder(rootCommand)
    // ...
    .UseExceptionHandler((ex, ctx) =>
    {
        var verbose = ctx.ParseResult.HasOption(GlobalOptions.Verbose);
        var writer = new ConsoleWriter(ctx.Console, consoleFormatProvider, verbose);

        if (verbose)
        {
            writer.Error(ex.ToString());
        }
        else
        {
            writer.Error(ex.Message);
        }
    })
    // ...
    .Build();
```

**Actual Behavior**:

The rule flags the arrow function as being formatted incorrectly. The code fix removes the 4 space indent on the lines after the arrow but the rule still flags the block of code as being incorrectly formatted.

If I wrap the block of code in braces the rule no longer flags this piece of code which is why I believe it's related to using top-level statements.

```csharp
{
    var parser = new CommandLineBuilder(rootCommand)
        // ...
        .UseExceptionHandler((ex, ctx) =>
        {
            var verbose = ctx.ParseResult.HasOption(GlobalOptions.Verbose);
            var writer = new ConsoleWriter(ctx.Console, consoleFormatProvider, verbose);

            if (verbose)
            {
                writer.Error(ex.ToString());
            }
            else
            {
                writer.Error(ex.Message);
            }
        })
        // ...
        .Build();
}
```

Something else I noticed is the error in my build output says `RCS0053: Fix formatting of arguments` while the docs say [`RCS0053: Fix formatting of a list`](https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS0053.md).

**Expected Behavior**:

Given the config being used I'd expect the formatting of this piece of code to not flag any formatting rules.

**Comment by JosefPihrt** [comment]
Hi,

thanks for the report. I can reproduce this bug and I have a fix for it.

The name "Fix formatting of a list" is general name. In this case it is a list of arguments but It also could be a list of parameters, attributes etc.

**Comment by fgimian** [comment]
Hey @JosefPihrt, was this part of the 4.1.1 release?  I started experiencing this issue when switching to top level statements with Roslynator 4.1.1.

![image](https://user-images.githubusercontent.com/1811813/187019080-1179dcba-db65-4152-b725-bada936b87c1.png)

Thanks heaps
Fotis

**Comment by JosefPihrt** [comment]
Hi,

Sorry for late answer.

It would be better to open new issue because these comments to closed issue can easily fall into oblivion.

Anyway, it was part if the 4.1.1 release. Unfortunately, you discovered another bug 😄.

**Comment by fgimian** [comment]
Thank you so much.  I'll be sure to raise a new issue in the future for such cases, apologies. 😄
